cmd/qat_plugin/README.md

```diff
@@ -1,39 +1,215 @@
+of this component are available on the Docker hub. These images are automatically built and uploaded
+from the latest master branch of this repository.
+
+Release tagged images of the components are also available on the Docker hub, tagged with their
+release version numbers.
+
+The deployment YAML files supplied with this component in this repository use the images with the `devel`
+tag by default. If you do not build your own local images, your Kubernetes cluster may pull down
+the devel images from the Docker hub by default.
+
+To use the release tagged versions of the images, edit the
+[YAML deployment files](../../deployments/qat_plugin/)
+appropriately.
+
+## Prerequisites
+
+The component has the same basic dependancies as the
+[generic plugin framework dependencies](../../README.md#about).
+
+You will also need [appropriate hardware installed](#checking-for-hardware).
+
+The Intel QuickAssist Technology software for Linux must be installed and configured.
+For more information, refer to the
+[Intel QuickAssist Technology Software for Linux - Getting Started Guide](https://01.org/sites/default/files/downloads/intelr-quickassist-technology/336212qatswgettingstartedguiderev003.pdf)
+
+For more information, refer to the
+[Intel QuickAssist Technology site on 01.org](https://01.org/intel-quickassist-technology)
+
+The demonstrations have their own requirements, listed in their own specific sections.
+
+## Getting the source code:
+
+```bash
+$ go get -d -u https://github.com/intel/intel-device-plugins-for-kubernetes
 ```
 
-### Build QAT device plugin:
+## Verify node kubelet config
+
+Every node that will be running the plugin must have the
+[kubelet device-plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)
+configured. For each node, check that the kubelet device plugin socket exists:
+
+```bash
+$ ls /var/lib/kubelet/device-plugins/kubelet.sock
+/var/lib/kubelet/device-plugins/kubelet.sock
 ```
+
+## Deploying as a DaemonSet
+
+To deploy the plugin as a DaemonSet, you first need to build a container image for the plugin and
+ensure that is visible to your nodes.
+
+### Build the plugin image
+
+The following will use `docker` to build a local container image called `intel/intel-qat-plugin`
+with the tag `devel`. The image build tool can be changed from the default docker by setting the
+`BUILDER` argument to the [Makefile](../../Makefile).
+
+```bash
 $ cd $GOPATH/src/github.com/intel/intel-device-plugins-for-kubernetes
-$ make qat_plugin
+$ make intel-qat-plugin
+...
+Successfully tagged intel/intel-qat-plugin:devel
 ```
 
-### Verify kubelet socket exists in /var/lib/kubelet/device-plugins/ directory:
+> **Note**: `kerneldrv` mode is excluded from the build by default. Add `BUILDTAGS=kerneldrv` to `make`
+> to get `kerneldrv` functionality added to the build.
+>
+> **FIXME**: Is this not counter-intuitive, given that the default YAML uses that mode?
```

Comment:
there's still a YAML for this but it's not the "default" one and we don't document it here.

Reply:
thx. done.